rclcpp/test/rclcpp/executors/test_executors.cpp

```diff
@@ -839,3 +839,33 @@ TEST(TestExecutors, testSpinUntilFutureCompleteNodePtr)
 
   rclcpp::shutdown();
 }
+
+// Check spin functions with non default context
+TEST(TestExecutors, testSpinWithNonDefaultContext)
+{
+  auto non_default_context = std::make_shared<rclcpp::Context>();
+  non_default_context->init(0, nullptr);
+
+  {
```

Comment:
What we want to test in this particular scope is, these call should work after shutting down the default context to verify those do not rely on the default context? current test is good enough just to makes sure it takes specified node base, but this did not fail before the PR either?

Reply:
This test works regardless of whether the default context is up or down, and it failed before PR as below.

```
Running main() from /opt/ros/rolling/src/gtest_vendor/src/gtest_main.cc
/home/hans/workspace/rclcpp_ws/src/rclcpp/rclcpp/test/rclcpp/executors/test_executors.cpp:852: Failure
Expected: rclcpp::spin_some(node->get_node_base_interface()) doesn't throw an exception.
  Actual: it throws rclcpp::exceptions::RCLInvalidArgument with description "failed to create guard condition: context argument is null, at ./src/rcl/guard_condition.c:65".
/home/hans/workspace/rclcpp_ws/src/rclcpp/rclcpp/test/rclcpp/executors/test_executors.cpp:854: Failure
Expected: rclcpp::spin_all(node->get_node_base_interface(), 1s) doesn't throw an exception.
  Actual: it throws rclcpp::exceptions::RCLInvalidArgument with description "failed to create guard condition: context argument is null, at ./src/rcl/guard_condition.c:65".
/home/hans/workspace/rclcpp_ws/src/rclcpp/rclcpp/test/rclcpp/executors/test_executors.cpp:866: Failure
Expected: check_spin_until_future_complete() doesn't throw an exception.
  Actual: it throws rclcpp::exceptions::RCLInvalidArgument with description "failed to create guard condition: context argument is null, at ./src/rcl/guard_condition.c:65".
```

But as you said, the environment with default context enabled is not suit for this test.
Should I move this test from TestExecutor suite to another or new Suite?
Or, to test in TestExecutor suite, should I shutdown default context in the test? (In this case, perhaps it is necessary to take care not to crash TestExecutors::TearDown)